src/main/java/at/hannibal2/skyhanni/features/event/hoppity/HoppityAPI.kt

```diff
@@ -0,0 +1,170 @@
+    private var hoppityEggChat = mutableListOf<String>()
+    private var duplicate = false
+    private var lastRarity = ""
+    private var lastName = ""
+    private var newRabbit = false
+    private var lastChatMeal: HoppityEggType? = null
+    private var lastDuplicateAmount: Long? = null
+    private var rabbitBought = false
+
+    private fun resetChatData() {
+        this.hoppityEggChat = mutableListOf()
+        this.duplicate = false
+        this.newRabbit = false
+        this.lastRarity = ""
+        this.lastName = ""
+        this.lastChatMeal = null
+        this.lastDuplicateAmount = null
+        this.rabbitBought = false
+    }
+
+    fun isHoppityEvent() = (SkyblockSeason.currentSeason == SkyblockSeason.SPRING || SkyHanniMod.feature.dev.debug.alwaysHoppitys)
+
+    /**
+     * REGEX-TEST: §f1st Chocolate Milestone
+     * REGEX-TEST: §915th Chocolate Milestone
+     * REGEX-TEST: §622nd Chocolate Milestone
+     */
+    private val milestoneNamePattern by ChocolateFactoryAPI.patternGroup.pattern(
+        "rabbit.milestone",
+        "(?:§.)*?(?<milestone>\\d{1,2})[a-z]{2} Chocolate Milestone",
+    )
+
+    /**
+     * REGEX-TEST: §7Reach §6300B Chocolate §7all-time to
+     * REGEX-TEST: §7Reach §61k Chocolate §7all-time to unlock
+     */
+    private val allTimeLorePattern by ChocolateFactoryAPI.patternGroup.pattern(
+        "milestone.alltime",
+        "§7Reach §6(?<amount>[\\d.MBk]*) Chocolate §7all-time.*",
+    )
+
+    /**
+     * REGEX-TEST: §7Spend §6150B Chocolate §7in the
+     * REGEX-TEST: §7Spend §62M Chocolate §7in the §6Chocolate
+     */
+    private val shopLorePattern by ChocolateFactoryAPI.patternGroup.pattern(
+        "milestone.shop",
+        "§7Spend §6(?<amount>[\\d.MBk]*) Chocolate §7in.*",
+    )
+
+    fun fireSideDishMessage() {
+        LorenzChatEvent(
+            "§d§lHOPPITY'S HUNT §r§dYou found a §r§6§lSide Dish §r§6Egg §r§din the Chocolate Factory§r§d!",
+            ChatComponentText(""),
+        ).postAndCatch()
+    }
+
+    @SubscribeEvent(priority = EventPriority.HIGH)
+    fun onSlotClick(event: GuiContainerEvent.SlotClickEvent) {
+        val index = event.slot?.slotIndex ?: return
+        if (index == -999) return
+
+        val clickedStack = InventoryUtils.getItemsInOpenChest()
+            .find { it.slotNumber == event.slot.slotNumber && it.hasStack }
+            ?.stack ?: return
+        val nameText = (if (clickedStack.hasDisplayName()) clickedStack.displayName else clickedStack.itemName)
+
+        milestoneNamePattern.matchMatcher(nameText) {
+            val itemLore = clickedStack.getLore()
+            if (!itemLore.any { it == "§eClick to claim!" }) return
+
+            // Will never match both all time and shop patterns together
+            allTimeLorePattern.firstMatcher(clickedStack.getLore()) {
+                LorenzChatEvent(
+                    "§d§lHOPPITY'S HUNT §r§dYou claimed a §r§6§lChocolate Milestone Rabbit §r§din the Chocolate Factory§r§d!",
```

Comment:
why do we send the chat message here? if we want to impact code in other spots, please use functions/events.

Reply:
This was my way to leave the Hoppity chat compacting and the other chat parsers intact while still firing the new data - I didn't see a clean way to interface with both of the existing solutions at once while also using event fires. I can look into it more, but this was the functional way I was able to get both modules interfacing with the API cleanly.

Reply:
I see. For now, the solution is fine, but when we want to expand the Hoppity/chocolate feature in the future, we should start to rework all this. Maybe into an event system for chat, or something.